Compilation fails under KDE neon

Sorry, I'm not much of a programmer and just started with QT. I just opened the project with qtcreator and tried to run it, but I end up with the following compilation errors:

```
../openmw-mod-manager/TreeModModel.cpp: In member function ‘void TreeModModel::addMods(const QJsonArray&, QModelIndex&)’:
../openmw-mod-manager/TreeModModel.cpp:306:46: error: invalid initialization of non-const reference of type ‘QModelIndex&’ from an rvalue of type ‘QModelIndex’
             addMods(subModsArray, this->index(newRow, 0, parent));
                                              ^
../openmw-mod-manager/TreeModModel.cpp:286:6: note:   initializing argument 2 of ‘void TreeModModel::addMods(const QJsonArray&, QModelIndex&)’
 void TreeModModel::addMods(const QJsonArray& modsArray, QModelIndex& parent)
      ^
../openmw-mod-manager/TreeModModel.cpp: In member function ‘void TreeModModel::loadDataFromJson()’:
../openmw-mod-manager/TreeModModel.cpp:314:37: error: invalid initialization of non-const reference of type ‘QModelIndex&’ from an rvalue of type ‘QModelIndex’
     addMods(modsArray, QModelIndex());
                                     ^
../openmw-mod-manager/TreeModModel.cpp:286:6: note:   initializing argument 2 of ‘void TreeModModel::addMods(const QJsonArray&, QModelIndex&)’
 void TreeModModel::addMods(const QJsonArray& modsArray, QModelIndex& parent)
      ^
Makefile:1702: recipe for target 'TreeModModel.o' failed
```

Any help would be greatly appreciated.

--- a/.gitignore
+++ b/.gitignore
@@ -1,7 +1,7 @@
 
 #ignore thumbnails created by windows
 Thumbs.db
-#Ignore files build by Visual Studio
+#Ignore files built by Visual Studio
 *.obj
 *.exe
 *.pdb
@@ -27,3 +27,11 @@ obj/
 [Rr]elease*/
 _ReSharper*/
 [Tt]est[Rr]esult*
+#Ignore files built by qmake and make
+*.o
+moc_*
+ui_*
+Makefile
+.qmake.stash
+*.swp
+*.app

--- a/SettingsInterface.cpp
+++ b/SettingsInterface.cpp
@@ -35,7 +35,7 @@ void SettingsInterface::save()
 
 QVariant SettingsInterface::getSetting(const QString& key)
 {
-	return json.object()["settings"].toObject()[key];
+	return (QVariant) json.object()["settings"].toObject()[key];
 }
 
 void SettingsInterface::setSetting(const QString& key, const QString& value)

--- a/TreeModModel.cpp
+++ b/TreeModModel.cpp
@@ -283,7 +283,7 @@ bool TreeModModel::setHeaderData(int section, Qt::Orientation orientation, const
 	return result;
 }
 
-void TreeModModel::addMods(const QJsonArray& modsArray, QModelIndex& parent)
+void TreeModModel::addMods(const QJsonArray& modsArray, const QModelIndex& parent)
 {
 	foreach( QJsonValue mod, modsArray )
 	{

--- a/TreeModModel.h
+++ b/TreeModModel.h
@@ -52,7 +52,7 @@ public slots:
 	void updateConflictSelection(const QItemSelection& selected, const QItemSelection& deselected);
 
 private:
-	void addMods(const QJsonArray& modsArray, QModelIndex& parent);
+	void addMods(const QJsonArray& modsArray, const QModelIndex& parent);
 	void loadDataFromJson();
 	void saveDataToJson();
 	void saveDataToConfig();

--- a/WinMain.cpp
+++ b/WinMain.cpp
@@ -11,13 +11,20 @@ WinMain::WinMain(QWidget *parent) :
 	ui(new Ui::WinMain)
 {
 	ui->setupUi(this);
-
+	
 	// Get OpenMW config folder.
-	QString conficFolder = QStandardPaths::locate(QStandardPaths::DocumentsLocation, "My Games/OpenMW", QStandardPaths::LocateDirectory);
+	QString configFolder;
+	#if defined(Q_OS_MACOS)
+		configFolder = QStandardPaths::locate(QStandardPaths::ConfigLocation, "openmw", QStandardPaths::LocateDirectory);
+	#elif defined(Q_OS_WIN)
+		configFolder = QStandardPaths::locate(QStandardPaths::DocumentsLocation, "My Games/OpenMW", QStandardPaths::LocateDirectory);
+	#else
+		#error Unsupported OS
+	#endif
 
 	// Load configs.
-	settings = new SettingsInterface(conficFolder + "/mods.json");
-	openMWConfig = new OpenMWConfigInterface(conficFolder + "/openmw.cfg");
+	settings = new SettingsInterface(configFolder + "/mods.json");
+	openMWConfig = new OpenMWConfigInterface(configFolder + "/openmw.cfg");
 
 	// Set up mod view.
 	TreeModModel* model = new TreeModModel(settings, openMWConfig);

--- a/readme.markdown
+++ b/readme.markdown
@@ -25,3 +25,5 @@ This tool was inspired by [Wrye Mash](http://www.uesp.net/wiki/Tes3Mod:Wrye_Mash
 ## Source
 
 This tool suffers from major spaghetti code, as it was written as an introduction to Qt. I'm sorry.
+
+To compile run `qmake` followed by `make`.

